### PR TITLE
Fix nasdf

### LINKED ISSUE #2981: Delete `nasdf` from Nyxt sources
@Ambrevar could you provide a step-by-step plan for us to gracefully handle NASDF? I have the feeling that it takes more than simply:

- Delete it from Nyxt sources
- Bump the NASDF version in [ntemplate](https://github.com/atlas-engineer/ntemplate)
- Send a patch to Guix
- Add as a submodule

I'm very confused about some PRs that seem somewhat related to it.

Also could you please clarify who is working on what, and who is responsible for what? Let's please make use of assigning issue to people. Thanks.

### PR DESCRIPTION
**Pull request by aadcg**
# Description

Fixes #2981

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] Git hygiene:
  - I have pulled from master before submitting this PR
  - There are no merge conflicts.
- [x] I've added the new dependencies as:
  - ASDF dependencies,
  - Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [x] I have performed a self-review of my own code.
- [ ] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [x] Documentation:
  - All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - I have updated the existing documentation to match my changes.
  - I have commented my code in hard-to-understand areas.
  - I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
    - Changelog update should be a separate commit.
  - I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [x] Compilation and tests:
  - My changes generate no new warnings.
  - I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - I ran the tests locally (`(asdf:test-system :nyxt)` and `(asdf:test-system :nyxt/gi-gtk)`) and they pass.

### CODE REVIEW

**Review by aartaka — COMMENTED** [review]
Looks good to me. Untested due to build failures. Minor comments:
- The changes regarding the renderer system go beyond the (I suppose) purpose of the PR—removing NASDF dependency from the submodules/libraries/etc.
- Same goes for keymap and `make-instance` checking—these are nor related to NASDF, are they?
- It's funny that you call NASDF-deleting PRs "fix" 😃

File: libraries/nasdf/compilation-tests.lisp [hunk 1]
```diff
@@ -37,7 +37,7 @@ The logic is:
   requiring arguments.
 - If there's any other error raised by `typep', then TYPE-SPECIFIER is
   likely not a type."
-  (or (documentation type-specifier 'type)
+  (or (ignore-errors (documentation type-specifier 'type))
```
Comment (jmercouris) [inline libraries/nasdf/compilation-tests.lisp:40]: Finally.

**Comment by jmercouris** [comment]
Thank you for getting rid of the matrix of tests. This is a very welcome change. I see a lot of removed lines.

**Comment by jmercouris** [comment]
Well rather simplifying the matrix, to the platforms that matter for our purposes, currently.

**Pull request merged**